python/tchannel/tornado/request.py

```diff
@@ -56,16 +61,34 @@ def __init__(
         self.service = service
         self.tracing = tracing or Trace()
         # argstreams is a list of InMemStream/PipeStream objects
-        self.argstreams = argstreams
+        self.argstreams = argstreams or [InMemStream(),
+                                         InMemStream(),
+                                         InMemStream()]
         self.checksum = checksum
         self.id = id
         self.headers = headers or {}
         self.state = StreamState.init
         self.endpoint = ""
-        self.header = None
-        self.body = None
         self.scheme = scheme
 
+        self.is_streaming_request = self._is_streaming_request()
+        if not self.is_streaming_request:
+            self._copy_argstreams = [
+                self.argstreams[0].clone(),
+                self.argstreams[1].clone(),
+                self.argstreams[2].clone(),
+            ]
+
+    def rewind(self, id=None):
+        self.id = id
+        self.argstreams = [
+            self._copy_argstreams[0].clone(),
+            self._copy_argstreams[1].clone(),
```

Comment:
Why is `copy_argstreams` being cloned twice?

Also, the `_copy_argstreams` attribute won't be set if the request is a streaming request. This will fail with an AttributeError or something.